TLM/TLM/Manager/Impl/ExtPathManager.cs

```diff
@@ -19,6 +20,92 @@ public class ExtPathManager
             _spiral = spiral ?? throw new ArgumentNullException(nameof(spiral));
         }
 
+        /// <summary>
+        /// Finds a suitable path position for a walking citizen with the given world position.
+        /// If secondary lane constraints are given also checks whether there exists another lane that matches those constraints.
+        /// </summary>
+        /// <param name="pos">world position</param>
+        /// <param name="laneTypes">allowed lane types</param>
+        /// <param name="vehicleTypes">allowed vehicle types</param>
+        /// <param name="otherLaneTypes">allowed lane types for secondary lane</param>
+        /// <param name="otherVehicleTypes">other vehicle types for secondary lane</param>
+        /// <param name="allowTransport">public transport allowed?</param>
+        /// <param name="allowUnderground">underground position allowed?</param>
+        /// <param name="position">resulting path position</param>
+        /// <returns><code>true</code> if a position could be found, <code>false</code> otherwise</returns>
+        public static bool FindCitizenPathPosition(Vector3 pos,
+                                                   NetInfo.LaneType laneTypes,
+                                                   VehicleInfo.VehicleType vehicleTypes,
+                                                   NetInfo.LaneType otherLaneTypes,
+                                                   VehicleInfo.VehicleType otherVehicleTypes,
+                                                   bool allowTransport,
+                                                   bool allowUnderground,
+                                                   out PathUnit.Position position) {
+            position = default(PathUnit.Position);
+            float minDist = 1E+10f;
+            if (Instance.FindPathPositionWithSpiralLoop(
```

Comment:
add argument names, a very long call is easier to read this way
only exception if `arg name = parameter name`, then it only adds noise